tests/test_example_token.cdc

```diff
@@ -0,0 +1,157 @@
+import Test
+
+pub let blockchain = Test.newEmulatorBlockchain()
+pub let admin = blockchain.createAccount()
+pub let recipient = blockchain.createAccount()
+
+pub fun setup() {
+    blockchain.useConfiguration(Test.Configuration({
+        "FungibleTokenMetadataViews": admin.address,
+        "ExampleToken": admin.address
+    }))
+
+    var code = Test.readFile("../contracts/FungibleTokenMetadataViews.cdc")
+    var err = blockchain.deployContract(
+        name: "FungibleTokenMetadataViews",
+        code: code,
+        account: admin,
+        arguments: []
+    )
+    Test.expect(err, Test.beNil())
+
+    code = Test.readFile("../contracts/ExampleToken.cdc")
+    err = blockchain.deployContract(
+        name: "ExampleToken",
+        code: code,
+        account: admin,
+        arguments: []
+    )
+
+    Test.expect(err, Test.beNil())
+}
+
+pub fun testTokensInitialized() {
+    let typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.TokensInitialized")!
+
+    Test.assertEqual(1, blockchain.eventsOfType(typ).length)
+}
+
+pub fun testSetupRecipientAccount() {
+    let code = Test.readFile("../transactions/setup_account.cdc")
+    let tx = Test.Transaction(
+        code: code,
+        authorizers: [recipient.address],
+        signers: [recipient],
+        arguments: []
+    )
+
+    let result = blockchain.executeTransaction(tx)
+
+    Test.expect(result, Test.beSucceeded())
+}
+
+pub fun testMintTokens() {
+    let code = Test.readFile("../transactions/mint_tokens.cdc")
+    let tx = Test.Transaction(
+        code: code,
+        authorizers: [admin.address],
+        signers: [admin],
+        arguments: [recipient.address, 250.0]
+    )
+
+    let result = blockchain.executeTransaction(tx)
+
+    Test.expect(result, Test.beSucceeded())
+
+    var typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.TokensMinted")!
+    Test.assertEqual(1, blockchain.eventsOfType(typ).length)
+
+    typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.MinterCreated")!
+    Test.assertEqual(1, blockchain.eventsOfType(typ).length)
+
+    typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.TokensDeposited")!
+    Test.assertEqual(1, blockchain.eventsOfType(typ).length)
+}
+
+pub fun testTransferTokens() {
+    let code = Test.readFile("../transactions/transfer_tokens.cdc")
+    let tx = Test.Transaction(
+        code: code,
+        authorizers: [recipient.address],
+        signers: [recipient],
+        arguments: [50.0, admin.address]
+    )
+
+    let result = blockchain.executeTransaction(tx)
+
+    Test.expect(result, Test.beSucceeded())
+
+    var typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.TokensDeposited")!
+    Test.assertEqual(2, blockchain.eventsOfType(typ).length)
+
+    typ = CompositeType("A.01cf0e2f2f715450.ExampleToken.TokensWithdrawn")!
+    Test.assertEqual(1, blockchain.eventsOfType(typ).length)
```

Comment:
Can you add scripts here to check here if the recipient's balance increased by 50 and the sender's balance decreased by 50? Thank you!

Reply:
Nice catch :100:
I have added more assertions in this test case: https://github.com/onflow/flow-ft/pull/137/commits/d7a27dc816bf01cad2082ca4aeaec2b781e9d969#diff-39b918cf828753efa2720618fdc579fe01b8bec8251d8c42c039a7e6ae9ef7cdR137-R160